newIDE/app/scripts/create-new-theme.js

```diff
@@ -0,0 +1,70 @@
+let args = process.argv;
+let path = require('path');
+let fs = require('fs');
+
+let theme = args[2];
+if (!theme) {
+  console.error('Please enter a theme name');
+  process.exit(1);
+}
+
+if (!theme.toLowerCase().endsWith('theme')) {
+  theme = theme + 'Theme';
+}
+
+const themeId = theme.replace(/\s+/, '');
+
+
+const dir = path.resolve(__dirname, '../src/UI/Theme/', themeId);
+if (fs.existsSync(dir)) {
+  console.warn('Theme `' + theme + '` already exists');
+  process.exit(0);
+}
+
+// Create Theme Folder
+fs.mkdirSync(dir);
+console.log('Created Folder', dir);
+
+// Create Style Library
+console.log(path.resolve(__dirname, './theme-templates/theme.json'));
+fs.writeFileSync(
+  path.resolve(dir, 'theme.json'),
+  fs.readFileSync(path.resolve(__dirname, './theme-templates/theme.json')).toString(),
+);
+console.log('Created theme.json');
+
+// Create Theme Object
+fs.writeFileSync(
+  path.resolve(dir, 'index.js'),
+  fs.readFileSync(path.resolve(__dirname, './theme-templates/index.js')).toString().replace(/\$themeId/g, themeId),
+);
+console.log('Created index.js');
+
+// Add to themes.json
+let registry = JSON.parse(fs.readFileSync(path.resolve(dir, '../themes.json')).toString())
+  .concat({
+    id: themeId,
+    name: theme,
+  });
+
+fs.writeFileSync(
+  path.resolve(dir, '../themes.json'),
+  JSON.stringify(registry, null, 2),
+);
+console.log('Updated themes.json');
+
+const imports =
+  registry.map(it => `import ${it.id} from './${it.id}';`).join('\n');
+
+const themes =
+  registry.map(it => `    ['${it.name}']: ${it.id}`).join(',\n');
+
+// Re-write ThemeRegistry based on themes.json
+fs.writeFileSync(
+  path.resolve(dir, '../ThemeRegistry.js'),
+  imports + '\n\n/*eslint no-useless-computed-key: "off"*/\nexport const themes = {\n' + themes + '\n};'
+);
+console.log('Updated ThemeRegistry.js');
```

Comment:
Nitpicking: in the scripts I usually favor using shelljs for writing things out and usually add an emoji to quickly show that things are properly progressing: `shell.echo('✅ Updated ThemeRegistry.js with the new theme');`

Reply:
done ✅